cmd/soroban-rpc/internal/methods/util.go

```diff
@@ -0,0 +1,32 @@
+package methods
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+)
+
+func getProtocolVersion(
+	ctx context.Context,
+	ledgerEntryReader db.LedgerEntryReader,
+	ledgerReader db.LedgerReader,
+) (uint32, error) {
+	latestLedger, err := ledgerEntryReader.GetLatestLedgerSequence(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	// obtain bucket size
+	closeMeta, ok, err := ledgerReader.GetLedger(ctx, latestLedger)
+	if err != nil {
+		return 0, err
+	}
+	if !ok {
+		return 0, fmt.Errorf("missing meta for latest ledger (%d)", latestLedger)
+	}
+	if closeMeta.V != 1 {
+		return 0, fmt.Errorf("latest ledger (%d) meta has unexpected verion (%d)", latestLedger, closeMeta.V)
+	}
+	return uint32(closeMeta.V1.LedgerHeader.Header.LedgerVersion), nil
```

Comment:
You can replace this with `getBucketListSizeAndProtocolVersion(ctx, ledgerReader, latestLedger)`, right?

Reply:
I intentionally kept it like that because I dont want to add dependency of `simulate_transaction.go` in here

Initially what I had in mind was to have just one common method which can be utilized by `simulate_txn` and `version_info` file but refactored that based on this [comment](https://github.com/stellar/soroban-rpc/pull/198#discussion_r1664979754).